.github/workflows/tests.yml

```diff
@@ -0,0 +1,49 @@
+name: tests
+
+on:
+  push:
+  pull_request:
+  schedule:
+    - cron: '0 0 * * *'
+
+jobs:
+  tests:
+    runs-on: ubuntu-20.04
+
+    services:
+      redis:
+        image: redis
+        ports:
+          - 6379:6379
+        options: --entrypoint redis-server
```

Comment:
Why do we need Redis here?

Reply:
sorry, copy that from another 1st party template